and-bible/app/src/main/java/net/bible/android/view/activity/page/MainBibleActivity.java

```diff
@@ -117,30 +115,7 @@ public void onCreate(Bundle savedInstanceState) {
 		// force the screen to be populated
 		PassageChangeMediator.getInstance().forcePageUpdate();
 		refreshScreenKeepOn();
-		checkSdcardReadPermission();
-	}
-
-	private void checkSdcardReadPermission() {
-		boolean sdcardPermissionAsked = CommonUtils.getSharedPreferences().getBoolean(SDCARD_PERMISSON_ASKED, false);
-		boolean permissionGranted = ContextCompat.checkSelfPermission(BibleApplication.getApplication(), Manifest.permission.READ_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED;
-
-		if(!sdcardPermissionAsked && !permissionGranted) {
-			new AlertDialog.Builder(this)
-					.setMessage(R.string.sdcard_permission_query)
-					.setCancelable(false)
-					.setPositiveButton(R.string.yes, new DialogInterface.OnClickListener() {
-						public void onClick(DialogInterface dialog, int id) {
-							ActivityCompat.requestPermissions(MainBibleActivity.this,
-									new String[]{Manifest.permission.READ_EXTERNAL_STORAGE}, SDCARD_READ_REQUEST);
-						}
-					})
-					.setNegativeButton(R.string.no, new DialogInterface.OnClickListener() {
-						public void onClick(DialogInterface dialog, int id) {
-							// Do not ask this any more
-							CommonUtils.getSharedPreferences().edit().putBoolean(SDCARD_PERMISSON_ASKED, true).apply();
-						}
-					}).create().show();
-		}
+		requestSdcardPermission();
```

Comment:
I think we are getting a bit muddled here.  I intended that immediately the user toggles the read manual install setting on the appropriate permission is requested.  Then it will not need to be requested during start up.

To see an example of how to request sd card perm if a setting changes look at timeoutPref.setOnPreferenceChangeListener in the old SettingsActivity code (it has been removed as part of these changes).

btw I think it would also be safer to request read sd card read permission if the user selects ''Load document from zip" during initial install unless you have tested that and it is unnecessary.

Reply:
> I think we are getting a bit muddled here. I intended that immediately the user toggles the read manual install setting on the appropriate permission is requested. Then it will not need to be requested during start up.

It is done just that way. But it is possible to revoke permission (from system's app settings), and then we need to request that permission in the startup if user has selected request_sdcard_permission_pref in settings screen.

> btw I think it would also be safer to request read sd card read permission if the user selects ''Load document from zip" during initial install unless you have tested that and it is unnecessary.

I have tested that it does not need permissions. Loading document from zipi uses system intent and receives file via stream from URI (which is given by system) rather than reading directly from filesystem (see InstallZip.java:215-217 and :221).